### PR TITLE
feat: add GetMaxMatureBlockNumber func

### PR DESCRIPTION
**Pull request by shaojunda**

### CODE REVIEW

File: utils/util.go [hunk 1]
```diff
@@ -51,3 +55,123 @@ func RemoveCellOutputData(cellOutputData [][]byte, index int) [][]byte {
...
+		return blockNumber, nil
+	}
+}
+
+type rationalNumber struct {
+	// Numerator.
+	numer uint64
+	// Denominator.
+	denom uint64
+}
```
Comment (TheWaWaR) [inline utils/util.go:131]: Both `number` and `denom` is not enough with `uint64`, use `big.Int` instead.

Comment (TheWaWaR) [inline utils/util.go:131]: Replace `rationalNumber` with [`big.Rat`](https://golang.org/pkg/math/big/#Rat)

File: utils/util.go [hunk 2]
```diff
@@ -51,3 +55,123 @@ func RemoveCellOutputData(cellOutputData [][]byte, index int) [][]byte {
...
+		}
+		return tipEpochMN, cellbaseMaturityMN
+	}
+}
+
+func isTipEpochLessThanCellbaseMaturity(tipEpoch *types.EpochParams, cellbaseMaturity *types.EpochParams) bool {
+	tipEpochRN, cellbaseMaturityRN := epochRationalNum(tipEpoch, cellbaseMaturity)
+	if tipEpochRN.numer-cellbaseMaturityRN.numer < 0 {
+		return true
+	}
```
Comment (TheWaWaR) [inline utils/util.go:161]: Use `big.Rat` instead.

File: utils/util.go [hunk 3]
```diff
@@ -51,3 +55,123 @@ func RemoveCellOutputData(cellOutputData [][]byte, index int) [][]byte {
...
+	nodeInfo, err := client.LocalNodeInfo(ctx)
+	if err != nil {
+		return 0, 0, 0, err
+	}
+	nodeVersion := strings.Split(nodeInfo.Version, " (")
+	versionArr := strings.Split(nodeVersion[0], ".")
+	major, err := strconv.Atoi(versionArr[0])
+	minor, err := strconv.Atoi(versionArr[1])
+	patch, err := strconv.Atoi(versionArr[2])
+	return major, minor, patch, nil
```
Comment (TheWaWaR) [inline utils/util.go:176]:
```suggestion
	nodeVersion := strings.Split(nodeInfo.Version, " (")
	versionArr := strings.Split(nodeVersion[0], ".")
	major, err := strconv.Atoi(versionArr[0])
	minor, err := strconv.Atoi(versionArr[1])
	patch, err := strconv.Atoi(versionArr[2])
	return major, minor, patch, nil
```
Put above line in a standalone function and test it.

File: README.md [hunk 4]
```diff
@@ -509,24 +511,81 @@ func main() {
...
+	searchKey := &indexer.SearchKey{
+		Script: &types.Script{
+			CodeHash: types.HexToHash("0x9bd7e06f3ecf4be0f2fcd2188b23f1b9fcc88e5d4b65a8637b17723bbda3cce8"),
+			HashType: types.HashTypeType,
+			Args:     args,
+		},
+		ScriptType: indexer.ScriptTypeLock,
+		Filter: &indexer.CellsFilter{
+			BlockRange: &[2]uint64{0, maxMatureBlockNumber},
+		},
```
Comment (TheWaWaR) [inline README.md:569]: The filter is wrong, must only filter out cellbase, and cellbase not in genesis block.

File: utils/util.go (hunk 1 above)
Comment (shaojunda) [reply utils/util.go]: updated: https://github.com/nervosnetwork/ckb-sdk-go/pull/26/commits/baf0fec7a78e27e22d4b1ba3a0bab4b4cfcd56ed

File: utils/util.go (hunk 2 above)
Comment (shaojunda) [reply utils/util.go:141]: updated: https://github.com/nervosnetwork/ckb-sdk-go/pull/26/commits/baf0fec7a78e27e22d4b1ba3a0bab4b4cfcd56ed

File: utils/util.go (hunk 3 above)
Comment (shaojunda) [reply utils/util.go]: updated: https://github.com/nervosnetwork/ckb-sdk-go/pull/26/commits/02abcd57065ec513f5865bbf86f72397059e2fa6

File: README.md (hunk 4 above)
Comment (shaojunda) [reply README.md]: updated: https://github.com/nervosnetwork/ckb-sdk-go/pull/26/commits/54e135a6d653e269edd286953b04d65172a54dcc https://github.com/nervosnetwork/ckb-sdk-go/pull/26/commits/e0b1f74ab7df412bbfab762405f336220b830ab2

File: utils/util.go [hunk 5]
```diff
@@ -51,3 +57,115 @@ func RemoveCellOutputData(cellOutputData [][]byte, index int) [][]byte {
...
+	patch, err := strconv.Atoi(versionArr[2])
+	if err != nil {
+		return 0, 0, 0, err
+	}
+	return major, minor, patch, nil
+}
+
+// IsMature check if a cellbase live cell is mature
+func IsMature(cell *indexer.LiveCell, maxMatureBlockNumber uint64) bool {
+	return cell.TxIndex > 0 || cell.BlockNumber == 0 || cell.BlockNumber <= maxMatureBlockNumber
```
Comment (TheWaWaR) [inline utils/util.go:170]: What is `cell.TxIndex` here mean?

Comment (shaojunda) [reply utils/util.go:170]: same as here: https://github.com/nervosnetwork/ckb-indexer/blob/master/src/service.rs#L235

means transaction index in the block.

File: README.md [hunk 6]
```diff
@@ -509,24 +511,81 @@ func main() {
...
+	}
+	var cells []*indexer.LiveCell
+	for iterator.HasNext() {
+		liveCell, err := iterator.CurrentItem()
+		if err != nil {
+			log.Fatalf("get cell error: %v", err)
+		}
+		if utils.IsMature(liveCell, maxMatureBlockNumber) {
+		  cells = append(cells, liveCell)
+        }
```
Comment (TheWaWaR) [inline README.md:581]: Mix use of <space> and <tab>

**Review by TheWaWaR — APPROVED** [review]

File: README.md (hunk 6 above)
Comment (shaojunda) [reply README.md]: updated: https://github.com/nervosnetwork/ckb-sdk-go/pull/26/commits/b607158a9414746b250028d9f741d63af6d18808

**Review by duanyytop — APPROVED** [review]

**Pull request merged**